enhancements/ingress/global-admission-hsts.md

```diff
@@ -0,0 +1,631 @@
+* Routes with hosts that match the `domainPatterns` of the HSTS will be validated
+* Routes with hosts that match the `domainPatterns` and `namespaceSelector` of the HSTS will be validated
+* Routes that only match the `namespaceSelector` will not be validated
+* Routes that are validated will validate that the `maxAge` exists and falls within the range of the HSTS `maxAgePolicy`
+* Routes that are validated will validate that the `preloadPolicy` and `includeSubdomainPolicy` match the HSTS Policy if they exist
+
+#### Audit HSTS configurations on any cluster
+This proposal offers the cluster administrator a way to review the HSTS configurations guaranteed by the admission plugin.
+For example, to review the maxAge set for required HSTS Policies:
+```shell
+oc get clusteroperator/ingress -n openshift-ingress-operator -o jsonpath='{range .spec.requiredHSTSPolicies}{.spec.requiredHSTSPolicies.maxAgePolicy.largestMaxAge}{"\n"}{end}'
+```
+To review the annotations on all routes:
+````shell
+oc get route  --all-namespaces -o go-template='{{range .items}}Name: {{.metadata.name}} Annotations: {{.metadata.annotations}} {{"\n"}}{{end}}'
+
+Name: oauth-openshift Annotations: <no value>
+Name: console Annotations: map[openshift.io/host.generated:true haproxy.router.openshift.io/hsts_header:max-age=900;preload;includeSubdomains]
+Name: downloads Annotations: map[include.release.openshift.io/ibm-cloud-managed:true include.release.openshift.io/self-managed-high-availability:true include.release.openshift.io/single-node-developer:true]
+Name: canary Annotations: map[haproxy.router.openshift.io/balance:roundrobin openshift.io/host.generated:true]
+...
+````
+#### Predict HSTS enforcement on any cluster with a single configuration manifest
+This proposal offers the cluster administrator a way to configure and enforce HSTS configurations on routes by configuring
+the `Ingress` using templates that substitute only the for the difference between clusters.  For example, for
+a development cluster with domain `dev.abc.com` and production cluster with `prod.abc.com` domains, administrators can use a single
+`Ingress` configuration template that substitutes the domain:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: Ingress
+metadata:
+  name: cluster
+spec:
+  domain: abc.com
+  requiredHSTSPolicies:
+    - domainPatterns:
+      - ${DOMAIN}
+      maxAge: 
+        smallestMaxAge: 1
+        largestMaxAge: 31536000
+      preloadPolicy: "RequirePreload"
+      includeSubdomainsPolicy: "RequireIncludeSubdomains"
+```
+All routes that require HSTS will then use the same required Annotation on either cluster:
+````yaml
+apiVersion: v1
+kind: Route
+metadata:
+  annotations:
+    haproxy.router.openshift.io/hsts_header: max-age=31536000;preload;includeSubdomains
+````
+
+### User Stories
+
+#### As a cluster administrator, I want to verify HSTS globally, for all TLS routes in domain `foo.com`
```

Comment:
good example.  Can we use your `oc get route --all-namespaces -o go-template='{{range .items}}Name: {{.metadata.name}} Annotations: {{.metadata.annotations}} {{"\n"}}{{end}}'` command with a field selector on the domain to find policy violators?

Reply:
Added an example.